packages/components/src/components/SquiggleViewer/keyboardNav/focusedSqValue.ts

```diff
@@ -0,0 +1,57 @@
+import { SqValuePath } from "@quri/squiggle-lang";
+
+import { useViewerContext } from "../ViewerProvider.js";
+import { focusSqValueHeader, keyboardEventHandler } from "./utils.js";
+
+const validKeys = [
+  "ArrowDown",
+  "ArrowUp",
+  "ArrowLeft",
+  "ArrowRight",
+  "Enter",
+] as const;
+
+export function useFocusedSqValueKeyEvent(selected: SqValuePath) {
+  const { setFocused, itemStore, findNode } = useViewerContext();
+
+  function resetToRoot() {
+    setFocused(undefined);
+
+    // This timeout is a hack to make sure the header is focused after the reset
+    setTimeout(() => {
+      focusSqValueHeader(selected, itemStore);
+    }, 1);
+  }
+
+  return keyboardEventHandler(validKeys, {
+    ArrowDown: () => {
+      const newItem = findNode(selected)?.children()[0];
+      if (newItem) {
+        focusSqValueHeader(newItem.node.path, itemStore);
+      }
+    },
+    ArrowUp: () => {
+      const newItem = findNode(selected)?.parent();
+      if (newItem) {
+        if (newItem.isRoot()) {
+          resetToRoot();
+        } else {
+          setFocused(newItem.node.path);
+        }
+      }
+    },
+    ArrowLeft: () => {
+      const newPath = findNode(selected)?.prevSibling()?.node.path;
+      if (newPath) {
+        setFocused(newPath);
+      }
+    },
+    ArrowRight: () => {
+      const newPath = findNode(selected)?.nextSibling()?.node.path;
+      if (newPath) {
+        setFocused(newPath);
+      }
```

Comment:
The actions here are inverted compared to the unfocused mode: "Left" and "Right" go through siblings, and "Up" and "Down" go up/down the tree.

I wonder if we could make it inverted, to match the unfocused mode (especially if you agree with my suggestion on `unfocusedSqValue.ts` file):
- "Up" and "Down" could scroll through siblings (but never go into children)
- "Left" would go one level up the tree (which feels natural to me; the breadcrumbs are horizontal)
- "Right" would go into nested values, and you could navigate things there, or press "Left" to get back to the top focused value